src/clj/tic_tac_toe_clojure/core.clj

```diff
@@ -11,21 +11,43 @@
   ["0" "1" "2" "3" "4" "5" "6" "7" "8"])
 
 (defn format-board [board]
-  (with-out-str (println (string/join "\n" (re-seq #".{1,3}" (apply str board))))))
+  (apply str (flatten (for [part (partition 3 board)] (flatten (map vector part '(" | " " | " "\n---------\n")))))))
 
 (defn display-board [board]
   (print (format-board board)))
 
-(defn get-user-symbol []
-  (do (print "Please choose a symbol (X or O): ") (flush) (read-line)))
+(def select-position "Please choose a position between 0 and 8: ")
 
-(defn get-user-position []
- (Integer/parseInt (do (print "Please choose a position between 0 and 8: ") (flush) (read-line))))
+(defn prompt-user [prompt]
+  (do (print prompt) (flush) (read-line)))
+
+(defn get-human-position []
+  (Integer/parseInt (prompt-user select-position)))
+
+;(defn get-computer-position [board randomiser]
+;  (get (randomiser (filter (fn [[_ marker]] (= "_" marker)) (map-indexed vector board))) 0))
+
+(defn get-first-available-position [board]
+  (get (first (filter (fn [[_ marker]] (= "_" marker)) (map-indexed vector board))) 0))
+
+(defn get-player-position [board player-type]
+  (if (= "h" player-type) (get-human-position) (get-first-available-position board)))
 
 (def set-position assoc)
 
-(defn swap-player [user-symbol]
-  (if (= "X" user-symbol) "O" "X"))
+(def select-opponent "Please select an opponent (H for human or C for computer): ")
+
+(defn get-game-type [opponent]
+  (with-out-str(print opponent)))
+
+(defn swap-type [player-type]
+  (if (= "h" player-type) "c" "h"))
+
+(defn swap-player-type [game-type player-type]
```

Comment:
Are you using this function? I cannot see the difference between this function and `swap-type` but maybe it's because I don't know clojure and it confuses me...

Reply:
These were a victim of the weird merge - they've been deleted for real this time!